### PR TITLE
update(api-index): add all content

### PR DESCRIPTION
**Pull request by NathalieC**
* rest api
* engine api
* specific use cases for both

### CODE REVIEW

File: modules/ROOT/pages/api-index.adoc [hunk 1]
```diff
@@ -1,5 +1,166 @@
-= Bonita API
-:description: blabla
+= Bonita API 
+:description: There are 2 types of API in Bonita: REST and Engine (Java). Learn how to use them here.
```
Comment (alachambre) [inline modules/ROOT/pages/api-index.adoc]:
```suggestion
:description: Bonita provides REST and Java API's to interact with the Runtime. Learn how to use them here.
```

File: modules/ROOT/pages/api-index.adoc [hunk 2]
```diff
@@ -1,5 +1,166 @@
...
 
-- Java API + java doc
-- Rest API
+There are 2 types of API in Bonita: REST and Engine (Java). Learn how to use them here.
```
Comment (alachambre) [inline modules/ROOT/pages/api-index.adoc]:
```suggestion
Bonita provides REST and Java API's to interact with the runtime. Learn how to use them here.
```

File: modules/ROOT/pages/api-index.adoc [hunk 3]
```diff
@@ -1,5 +1,166 @@
...
+--
+
+[.card.card-index]
+--
+xref:api-glossary.adoc[[.card-title]#REST API glossary# [.card-body.card-content-overflow]#pass:q[A few things to know to best use the REST API]#]
+--
+
+[.card.card-index]
+--
+xref:rest-api-authentication.adoc[[.card-title]#REST Authentication# [.card-body.card-content-overflow]#pass:q[Login to and logout from Bonita, the prerequisites to call the REST API]#]
```
Comment (alachambre) [inline modules/ROOT/pages/api-index.adoc]:
```suggestion
xref:rest-api-authentication.adoc[[.card-title]#REST Authentication# [.card-body.card-content-overflow]#pass:q[Authentication is a prerequisite to any other REST API call]#]
```

File: modules/ROOT/pages/api-index.adoc [hunk 4]
```diff
@@ -1,5 +1,166 @@
...
+xref:rest-api-extensions.adoc[[.card-title]#REST API extensions overview# [.card-body.card-content-overflow]#pass:q[Introduction and example]#]
+--
+
+[.card.card-index]
+--
+xref:rest-api-extension-archetype.adoc[[.card-title]#Create a REST API extension# [.card-body.card-content-overflow]#pass:q[Tools to develop and test your REST API Extensions]#]
+--
+
+[.card-section]
+== REST API per project element
```
Comment (alachambre) [inline modules/ROOT/pages/api-index.adoc]:
```suggestion
== REST API - per theme
```

File: modules/ROOT/pages/api-index.adoc [hunk 5]
```diff
@@ -1,5 +1,166 @@
...
+--
+
+[.card.card-index]
+--
+xref:bdm-api.adoc[[.card-title]#BDM API# [.card-body.card-content-overflow]#pass:q[Learn how to query the Business data]#]
+--
+
+[.card.card-index]
+--
+xref:bpm-api.adoc[[.card-title]#BPM API# [.card-body.card-content-overflow]#pass:q[Learn how to retrieve activities and subprocesses, or to perform maintenance tasks]#]
```
Comment (alachambre) [inline modules/ROOT/pages/api-index.adoc]: Too long

```suggestion
xref:bpm-api.adoc[[.card-title]#BPM API# [.card-body.card-content-overflow]#pass:q[Learn how to retrieve activities and to perform maintenance tasks]#]
```

File: modules/ROOT/pages/api-index.adoc [hunk 6]
```diff
@@ -1,5 +1,166 @@
...
+== REST API per project element
+
+[.card.card-index]
+--
+xref:application-api.adoc[[.card-title]#Application API# [.card-body.card-content-overflow]#pass:q[Learn how to retrieve the current deployed application definitions]#]
+--
+
+[.card.card-index]
+--
+xref:access-control-api.adoc[[.card-title]#BDM Access Control API# [.card-body.card-content-overflow]#pass:q[Learn how to get the BDM Access Control status (lastUpdatedBy, lastUpdateDate…)]#]
```
Comment (alachambre) [inline modules/ROOT/pages/api-index.adoc]:
```suggestion
xref:access-control-api.adoc[[.card-title]#BDM Access Control API# [.card-body.card-content-overflow]#pass:q[Retrieve BDM Access Control status (lastUpdatedBy, lastUpdateDate…)]#]
```

File: modules/ROOT/pages/api-index.adoc [hunk 7]
```diff
@@ -1,5 +1,166 @@
...
+--
+
+[.card.card-index]
+--
+xref:platform-api.adoc[[.card-title]#Platform API# [.card-body.card-content-overflow]#pass:q[Learn how to manage Bonita platform-level resources]#]
+--
+
+[.card.card-index]
+--
+xref:portal-api.adoc[[.card-title]#Portal API# [.card-body.card-content-overflow]#pass:q[Learn how to access and manage pages, profiles, profile entries, profile members, and themes]#]
```
Comment (alachambre) [inline modules/ROOT/pages/api-index.adoc]: Way too long. Btw, what are profile entries?

```suggestion
xref:portal-api.adoc[[.card-title]#Portal API# [.card-body.card-content-overflow]#pass:q[Learn how to access and manage pages, profiles and themes]#]
```

File: modules/ROOT/pages/api-index.adoc [hunk 8]
```diff
@@ -1,5 +1,166 @@
...
+--
+
+[.card.card-index]
+--
+xref:system-api.adoc[[.card-title]#System API# [.card-body.card-content-overflow]#pass:q[Learn how to get informations on the current session and on localisation / translation.]#]
```
Comment (alachambre) [inline modules/ROOT/pages/api-index.adoc]: Too long. No s for information in plural form :p 

```suggestion
xref:system-api.adoc[[.card-title]#System API# [.card-body.card-content-overflow]#pass:q[Retrieve information on the current session and on localisation / translation.]#]
```

File: modules/ROOT/pages/api-index.adoc [hunk 9]
```diff
@@ -1,5 +1,166 @@
...
+--
+
+[.card.card-index]
+--
+xref:tenant-api.adoc[[.card-title]#Tenant API# [.card-body.card-content-overflow]#pass:q[Learn how to show the tenant status or install / update the BDM]#]
+--
+
+
+[.card-section]
+== Engine API - the basics
```
Comment (alachambre) [inline modules/ROOT/pages/api-index.adoc]: Both REST and JAVA API's are provided by the engine. Calling the Java API's 'engine API' is inaccurate and incomprehensible for developers.

```suggestion
== Java API - the basics
```

File: modules/ROOT/pages/api-index.adoc [hunk 10]
```diff
@@ -1,5 +1,166 @@
...
+
+[.card.card-index]
+--
+xref:engine-api-overview.adoc[[.card-title]#Overview# [.card-body.card-content-overflow]#pass:q[View the high-level list and usage of Engine API]#]
```
Comment (alachambre) [inline modules/ROOT/pages/api-index.adoc]:
```suggestion
xref:engine-api-overview.adoc[[.card-title]#Overview# [.card-body.card-content-overflow]#pass:q[View the high-level list and usage of Java API]#]
```

File: modules/ROOT/pages/api-index.adoc [hunk 11]
```diff
@@ -1,5 +1,166 @@
...
+--
+
+[.card.card-index]
+--
+https://javadoc.bonitasoft.com/api/{javadocVersion}/index.html[[.card-title]#Javadoc# [.card-body.card-content-overflow]#pass:q[View the complete list of Engine API with all the resources you need to use them efficiently]#]
```
Comment (alachambre) [inline modules/ROOT/pages/api-index.adoc]: To long

```suggestion
https://javadoc.bonitasoft.com/api/{javadocVersion}/index.html[[.card-title]#Javadoc# [.card-body.card-content-overflow]#pass:q[Complete Javadoc of the Java API]#]
```

File: modules/ROOT/pages/api-index.adoc [hunk 12]
```diff
@@ -1,5 +1,166 @@
...
+--
+
+[.card-section]
+== Common use-cases with Engine API
```
Comment (alachambre) [inline modules/ROOT/pages/api-index.adoc]:
```suggestion
== Common use-cases with Java API 
```

File: modules/ROOT/pages/api-index.adoc [hunk 13]
```diff
@@ -1,5 +1,166 @@
...
+--
+
+[.card.card-index]
+--
+xref:handle-a-failed-activity.adoc[[.card-title]#Handling a failed activity# [.card-body.card-content-overflow]#pass:q[Learn how to create process error recovery through Java API]#]
+--
+
+[.card.card-index]
+--
+xref:manage-an-organization.adoc[[.card-title]#Managing an organization# [.card-body.card-content-overflow]#pass:q[Learn how to manage people involved in the process execution through Java API]#]
```
Comment (alachambre) [inline modules/ROOT/pages/api-index.adoc]: To long

```suggestion
xref:manage-an-organization.adoc[[.card-title]#Managing an organization# [.card-body.card-content-overflow]#pass:q[Manage people involved in the process execution through Java API]#]
```

File: modules/ROOT/pages/api-index.adoc [hunk 14]
```diff
@@ -1,5 +1,166 @@
...
+--
+https://javadoc.bonitasoft.com/api/{javadocVersion}/index.html[[.card-title]#Javadoc# [.card-body.card-content-overflow]#pass:q[View the complete list of Engine API with all the resources you need to use them efficiently]#]
+--
+
+[.card-section]
+== Common use-cases with Engine API 
+
+[.card.card-index]
+--
+xref:manage-a-process.adoc[[.card-title]#Managing a process# [.card-body.card-content-overflow]#pass:q[Learn how to manage a process through Java API]#]
```
Comment (alachambre) [inline modules/ROOT/pages/api-index.adoc]:
```suggestion
xref:manage-a-process.adoc[[.card-title]#Managing a process# [.card-body.card-content-overflow]#pass:q[Manage processes through Java API]#]
```

File: modules/ROOT/pages/api-index.adoc [hunk 15]
```diff
@@ -1,5 +1,166 @@
...
+--
+
+[.card.card-index]
+--
+xref:handle-a-failed-activity.adoc[[.card-title]#Handling a failed activity# [.card-body.card-content-overflow]#pass:q[Learn how to create process error recovery through Java API]#]
```
Comment (alachambre) [inline modules/ROOT/pages/api-index.adoc]:
```suggestion
xref:handle-a-failed-activity.adoc[[.card-title]#Handling a failed activity# [.card-body.card-content-overflow]#pass:q[Create process error recovery through Java API]#]
```

File: modules/ROOT/pages/api-index.adoc [hunk 16]
```diff
@@ -1,5 +1,166 @@
...
+--
+
+[.card.card-index]
+--
+xref:manage-an-organization.adoc[[.card-title]#Managing an organization# [.card-body.card-content-overflow]#pass:q[Learn how to manage people involved in the process execution through Java API]#]
+--
+
+[.card.card-index]
+--
+xref:manage-users.adoc[[.card-title]#Managing users# [.card-body.card-content-overflow]#pass:q[Learn how to manage the users of your organization through Java API]#]
```
Comment (alachambre) [inline modules/ROOT/pages/api-index.adoc]:
```suggestion
xref:manage-users.adoc[[.card-title]#Managing users# [.card-body.card-content-overflow]#pass:q[Manage the users of your organization through Java API]#]
```

File: modules/ROOT/pages/api-index.adoc [hunk 17]
```diff
@@ -1,5 +1,166 @@
...
+--
+
+[.card.card-index]
+--
+xref:handling-documents.adoc[[.card-title]#Handling documents# [.card-body.card-content-overflow]#pass:q[Learn how to handle documents within the process through Java API]#]
```
Comment (alachambre) [inline modules/ROOT/pages/api-index.adoc]:
```suggestion
xref:handling-documents.adoc[[.card-title]#Handling documents# [.card-body.card-content-overflow]#pass:q[Handle documents within processes through Java API]#]
```

File: modules/ROOT/pages/api-index.adoc [hunk 18]
```diff
@@ -1,5 +1,166 @@
...
+--
+
+[.card.card-index]
+--
+xref:manage-the-platform.adoc[[.card-title]#Managing the platform# [.card-body.card-content-overflow]#pass:q[Learn how to manage the platform through Java API]#]
```
Comment (alachambre) [inline modules/ROOT/pages/api-index.adoc]:
```suggestion
xref:manage-the-platform.adoc[[.card-title]#Managing the platform# [.card-body.card-content-overflow]#pass:q[Manage the platform through Java API]#]
```

File: modules/ROOT/pages/api-index.adoc [hunk 19]
```diff
@@ -1,5 +1,166 @@
...
+--
+
+[.card.card-index]
+--
+xref:manage-files-using-upload-servlet-and-rest-api.adoc[[.card-title]#Manage files using upload servlet and REST API# [.card-body.card-content-overflow]#pass:q[Learn how to manage files through REST API]#]
+--
+
+[.card.card-index]
+--
+xref:create-administration-tools.adoc[[.card-title]#Creating administration tools# [.card-body.card-content-overflow]#pass:q[Learn how to create your onwn tools to administrate the platform through Java API]#]
```
Comment (alachambre) [inline modules/ROOT/pages/api-index.adoc]:
```suggestion
xref:create-administration-tools.adoc[[.card-title]#Create administration tools# [.card-body.card-content-overflow]#pass:q[Learn how to create your own tools to administrate the platform through Java API]#]
```

**Review by alachambre — CHANGES_REQUESTED** [review]
Do not talk about Engine API. 
Both REST AND JAVA API's are provided by the Engine, it is inaccurate and incomprehensible for developers to call Java API _'Engine API'_. 
Please update the cards, the menu name and any reference to the _'engine API'_ you could find in the documentation if possible.

File: modules/ROOT/pages/api-index.adoc (hunk 7 above)
Comment (NathalieC) [reply modules/ROOT/pages/api-index.adoc]: The "Portal API" part will be handled by the Runtime team, as @pabloalonso told me.

File: modules/ROOT/pages/api-index.adoc [hunk 20]
```diff
@@ -1,5 +1,171 @@
...
+=== Per theme
+
+[.card.card-index]
+--
+xref:application-api.adoc[[.card-title]#Application API# [.card-body.card-content-overflow]#pass:q[Retrieve the current deployed application definitions]#]
+--
+
+[.card.card-index]
+--
+xref:access-control-api.adoc[[.card-title]#BDM Access Control API# [.card-body.card-content-overflow]#pass:q[Retrieve BDM Access Control status (lastUpdatedBy, lastUpdateDate…)]#]
```
Comment (alachambre) [inline modules/ROOT/pages/api-index.adoc]: Too long for the card

```suggestion
xref:access-control-api.adoc[[.card-title]#BDM Access Control API# [.card-body.card-content-overflow]#pass:q[Retrieve BDM Access Control status]#]
```

File: modules/ROOT/pages/api-index.adoc [hunk 21]
```diff
@@ -1,5 +1,171 @@
...
+--
+
+[.card.card-index]
+--
+xref:bdm-api.adoc[[.card-title]#BDM API# [.card-body.card-content-overflow]#pass:q[Query the Business data]#]
+--
+
+[.card.card-index]
+--
+xref:bpm-api.adoc[[.card-title]#BPM API# [.card-body.card-content-overflow]#pass:q[Retrieve activities and to perform maintenance tasks]#]
```
Comment (alachambre) [inline modules/ROOT/pages/api-index.adoc]:
```suggestion
xref:bpm-api.adoc[[.card-title]#BPM API# [.card-body.card-content-overflow]#pass:q[Retrieve activities, perform maintenance tasks]#]
```

**Review by alachambre — CHANGES_REQUESTED** [review]
Still a few things to improve and I think we are good for this one :)

**Pull request merged**